packages/web/components/animation/dynamic-lottie-animation.tsx

```diff
@@ -0,0 +1,25 @@
+import dynamic from "next/dynamic";
+import { ComponentProps, FunctionComponent, useEffect, useState } from "react";
+
+const Lottie = dynamic(() => import("lottie-react"), { ssr: false });
+
+/** Loads the lottie libarary component, and it's
+ *  animation data dynamically (outside of the main bundle). */
+export const DynamicLottieAnimation: FunctionComponent<
+  {
+    /** JSON file source, likely a dynamic import: `import("./lottie.json")`. */
+    futureLottieContents: Promise<any>;
+  } & Omit<ComponentProps<typeof Lottie>, "animationData">
+> = (props) => {
+  // JSON contents
+  const [lottie, setLottie] = useState<any | null>(null);
+
+  // dynamic load JSON animation data
+  useEffect(() => {
+    if (!lottie) {
+      props.futureLottieContents.then(setLottie);
```

Comment:
is it possible that there would be an error here?

Reply:
There may be an error. Do you have an idea of how it should be handled? Errors are not expected and therefore can bubble up as an uncaught exception (in promise).